ql/src/java/org/apache/hadoop/hive/ql/exec/vector/VectorizedBatchUtil.java

```diff
@@ -999,4 +1002,16 @@ public static void debugDisplayBatch(VectorizedRowBatch batch, String prefix) {
       debugDisplayOneRow(batch, index, prefix);
     }
   }
+
+  /**
+   * Reset all columns other than Partition columns
+   * @param rowBatch VectorizedRowBatch to reset
+   */
+  public static void resetNonPartitionColumns(VectorizedRowBatch rowBatch, VectorizedRowBatchCtx context) {
```

Comment:
no need for context, VectorizedRowBatch already has getDataColumnCount() & getPartitionColumnCount()

Reply:
Ahh, my bad. Wasn't sure if the values would have got modified due to some Vector reuse. Didn't realize vectorRowBatch gets created based on the context. Fixed it.